mobile/native/src/ln_dlc/channel_status.rs

```diff
@@ -104,100 +93,63 @@ async fn channel_status(node: impl Borrow<Node>) -> Result<ConcreteChannelStatus
 #[derive(Debug, Clone, Copy)]
 enum ConcreteChannelStatus {
     NotOpen,
-    LnOpen,
-    LnDlcOpen,
-    // We cannot easily model the vanilla Lightning channel being force-closed because LDK erases
-    // a closed channel from the `ChannelManager` as soon as the closing process begins. This is a
-    // problem because it's not safe to delete the app if the channel is not fully closed!
-    //
-    // TODO: Add support for `LnForceClosing` status and ensure that channel status remains in
-    // `LnDlcForceClosing` whilst the LN channel is still closing.
-    // LnForceClosing,
-    LnDlcForceClosing,
-    Inconsistent,
+    Pending,
+    Open,
+    WithPosition,
+    Closing,
 }
 
-// TODO: We currently only look at the state of the subchannel. We should also take into account the
-// state of the DLC channel and the contract. Otherwise we won't be able to convey that the CET is
-// not yet confirmed.
-//
-// TODO: We should map the argument types to our own types so that we can simplify them. This will
-// make it so much easier to build test cases in order to add tests.
-fn derive_ln_dlc_channel_status(
-    // The most recently created LN channel, whether open or not.
-    ln_channel: Channel,
-    // All the subchannels that we have ever recorded.
-    subchannels: Vec<SubChannel>,
-) -> ConcreteChannelStatus {
-    match ln_channel {
-        // If the LN channel is open (or practically open).
-        Channel {
-            channel_id: Some(channel_id),
-            channel_state: ChannelState::Pending | ChannelState::Open | ChannelState::OpenUnpaid,
+fn derive_dlc_channel_status(dlc_channel: Option<&SignedChannel>) -> ConcreteChannelStatus {
```

Comment:
Given that we take an `Option<&SignedChannel>` I think that means that we consider a new channel that is not yet `Established` `NotOpen`. That seems fine, even if it would be nice to track the state of the opening protocol too.

But that would require looking at `Channel`s beyond `SignedChannel`, which is a bit trickier.

Reply:
I've simplified this part and replicate the libraries states in our code.